services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -1924,6 +1940,26 @@ export class WorkflowService {
     return documentsWithPersistedImages;
   }
 
+  async updateDocumentProperties(
+    workflowId: string,
+    targetDocumentId: string,
+    properties: object,
+    projectIds: TProjectIds,
+  ) {
+    const workflow = await this.workflowRuntimeDataRepository.findById(workflowId, {}, projectIds);
+
+    workflow.context.documents?.map((document: any) => {
```

Comment:
1. You don't use the map value. and pass the existing context from the database to be merged - so it wont change anything
2. If you're updating a single document you don't have to keep the other document, just the document you want to update
3. Error handling: Youre depending on array update by id, make sure there is an id
4. be explicit when calling the update method and pass 'by-id' param